relay-monitors/src/lib.rs

```diff
@@ -160,6 +167,10 @@ pub fn process_check_in(payload: &[u8]) -> Result<Vec<u8>, ProcessCheckInError>
         return Err(ProcessCheckInError::EmptySlug);
     }
 
+    if check_in.environment.len() > ENVIRONMENT_LENGTH {
```

Comment:
Here I think you want the number of chars, not the number of bytes. Rust is nasty.

```suggestion
    if check_in.environment.chars().count() > ENVIRONMENT_LENGTH {
```